apps/flutter_parent/lib/screens/events/event_details_screen.dart

```
@@ -0,0 +1,193 @@
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+import 'package:flutter/material.dart';
+import 'package:flutter_parent/l10n/app_localizations.dart';
+import 'package:flutter_parent/models/schedule_item.dart';
+import 'package:flutter_parent/screens/events/event_details_interactor.dart';
+import 'package:flutter_parent/utils/common_widgets/constrained_web_view.dart';
+import 'package:flutter_parent/utils/common_widgets/error_panda_widget.dart';
+import 'package:flutter_parent/utils/common_widgets/loading_indicator.dart';
+import 'package:flutter_parent/utils/service_locator.dart';
+import 'package:intl/intl.dart';
+
+class EventDetailsScreen extends StatefulWidget {
+  final ScheduleItem event;
+  final String eventId;
+
+  EventDetailsScreen.withEvent({Key key, this.event})
+      : assert(event != null),
+        eventId = event.id,
+        super(key: key);
+
+  EventDetailsScreen.withId({Key key, this.eventId})
+      : assert(eventId != null),
+        event = null,
+        super(key: key);
+
+  @override
+  _EventDetailsScreenState createState() => _EventDetailsScreenState();
+}
+
+class _EventDetailsScreenState extends State<EventDetailsScreen> {
+  GlobalKey<RefreshIndicatorState> _refreshKey = GlobalKey<RefreshIndicatorState>();
+
+  Future<ScheduleItem> _eventFuture;
+
+  Future<ScheduleItem> _loadEvent({bool forceRefresh = false}) => _interactor.loadEvent(widget.eventId, forceRefresh);
+
+  EventDetailsInteractor get _interactor => locator<EventDetailsInteractor>();
+
+  @override
+  void initState() {
+    if (widget.event != null) {
+      _eventFuture = Future.value(widget.event);
+    } else {
+      _eventFuture = _loadEvent();
+    }
+    super.initState();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return RefreshIndicator(
+      key: _refreshKey,
+      onRefresh: () {
```

I did a pull to refresh and the refresh indicator isn't going away.

---

LOL, this was because I never registered the interactor in the service locator, so it was just silently failing (passing in tests as I had it mocked there).

---

#oof